endpoints-framework/src/main/java/com/google/api/server/spi/config/model/EndpointsFlag.java

```diff
@@ -41,36 +34,48 @@
  * <li>falsy is defined as a case-insensitive equality with "false".</li>
  * </ul>
  */
-public enum MapSchemaFlag {
+public enum EndpointsFlag {
 
   /**
    * Reenabled the previous behavior of Cloud Endpoints, using untyped "JsonMap" for all Map types.
+   * By default, schema generation uses "additionalProperties" in JsonSchema to describe Map types
+   * (both for Discovery and OpenAPI), with a proper description of the value types.
    */
-  FORCE_JSON_MAP_SCHEMA("forceJsonMapSchema"),
+  MAP_SCHEMA_FORCE_JSON_MAP_SCHEMA("mapSchema.forceJsonMapSchema"),
 
   /**
    * When enabled, schema generation will not throw an error when handling Map types with keys that
    * are not serializable from / to string (previous Cloud Endpoints behavior). It will still
-   * probably generate an error when serializing / deserializing these types at runtime.
+   * probably generate an error when serializing / deserializing these types at runtime. {@link
+   * #MAP_SCHEMA_FORCE_JSON_MAP_SCHEMA} must be enabled for this to take effect.
    */
-  IGNORE_UNSUPPORTED_KEY_TYPES("ignoreUnsupportedKeyTypes"),
+  MAP_SCHEMA_IGNORE_UNSUPPORTED_KEY_TYPES("mapSchema.ignoreUnsupportedKeyTypes"),
 
   /**
    * Array values in "additionalProperties" are supported by the API Explorer, but not by the Java
    * client generation. This flag can be enabled when deploying an API to the server, but should
-   * always be disabled when generating Java clients.
+   * always be disabled when generating Java clients. {@link #MAP_SCHEMA_FORCE_JSON_MAP_SCHEMA} must
+   * be enabled for this to take effect.
    */
-  SUPPORT_ARRAYS_VALUES("supportArrayValues");
+  MAP_SCHEMA_SUPPORT_ARRAYS_VALUES("mapSchema.supportArrayValues"),
 
-  private static final String ENV_VARIABLE_PREFIX = "ENDPOINTS_MAP_SCHEMA_";
-  private static final String SYSTEM_PROPERTY_PREFIX = "endpoints.mapSchema.";
+  /**
+   * When enabled, allows use of Jackson serialization annotations. Previously, the Jackson
+   * annotation introspector was unused because Jackson was a vendored dependency. Now that Jackson
+   * is an explicit dependency, this can cause conflict with apps that use Jackson annotations for
+   * reasons outside of using this framework.
+   */
+  JSON_DISABLE_JACKSON_ANNOTATIONS("json.disableJacksonAnnotations");
```

Comment:
The comment says that enabling this option allows the use of annotations; that doesn't make much sense when compared to the name of the option.

Reply:
Awkward. I had the flag flipped the other way around, until I realized I had to invert it to have the correct default behavior.